cdp/smart_contract.py

```diff
@@ -0,0 +1,324 @@
+import json
+import time
+from decimal import Decimal
+from typing import Any
+
+from eth_account.signers.local import LocalAccount
+
+from cdp.cdp import Cdp
+from cdp.client.models.create_smart_contract_request import CreateSmartContractRequest
+from cdp.client.models.deploy_smart_contract_request import DeploySmartContractRequest
+from cdp.client.models.smart_contract import SmartContract as SmartContractModel
+from cdp.client.models.smart_contract_type import SmartContractType as SmartContractTypeModel
+from cdp.transaction import Transaction
+
+
+class SmartContract:
+    """A representation of a SmartContract on the blockchain."""
+
+    class Type:
+        """Enumeration of SmartContract types."""
```

Comment:
Let's make this a child class of `Enum` from `from enum import Enum` like we do in other classes with enumerated subclasses (E.x. `Transaction.Status`)